### PR TITLE
Madara: add some more sites

### LINKED ISSUE #14835: Hentai3z: A Branch of Hentai20, maybe?
### Source name

Hentai3z

### Source link

https://hentai3z.io

### Language

English

### Other details

18+/NSFW = yes

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a title with source name.
- [X] I have checked that the extension does not already exist on the [website extensions list](https://tachiyomi.org/extensions/) or the app.
- [X] I have checked that the extension does not already exist by searching the [GitHub repository](https://github.com/tachiyomiorg/tachiyomi-extensions/) and verified it does not appear in the code base.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #15277: Manhuafast.net
### Source name

ManhuaFast

### Source link

https://manhuafast.net/

### Language

English

### Other details

there is another source for Manhuafast.com but they are two different websites with different release and schedules, even though they host similar series.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a title with source name.
- [X] I have checked that the extension does not already exist on the [website extensions list](https://tachiyomi.org/extensions/) or the app.
- [X] I have checked that the extension does not already exist by searching the [GitHub repository](https://github.com/tachiyomiorg/tachiyomi-extensions/) and verified it does not appear in the code base.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #15788: Hentai3z
### Source name

Hentai3z

### Source link

https://hentai3z.cc/

### Language

English

### Other details

18+/NSFW=yes

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a title with source name.
- [X] I have checked that the extension does not already exist on the [website extensions list](https://tachiyomi.org/extensions/) or the app.
- [X] I have checked that the extension does not already exist by searching the [GitHub repository](https://github.com/tachiyomiorg/tachiyomi-extensions/) and verified it does not appear in the code base.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #16224: Jimanga
### Source name

jimanga 

### Source link

https://jimanga.com

### Source language

english

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a title with source name.
- [X] I have checked that the extension does not already exist on the [website extensions list](https://tachiyomi.org/extensions/) or the app.
- [X] I have checked that the extension does not already exist by searching the [GitHub repository](https://github.com/tachiyomiorg/tachiyomi-extensions/) and verified it does not appear in the code base.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #16581: v2.comiz.net
### Source name

Comiz 

### Source link

https://v2.comiz.net/

### Source language

English

### Other details

"18+/NSFW = yes"

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a title with source name.
- [X] I have checked that the extension does not already exist on the [website extensions list](https://tachiyomi.org/extensions/) or the app.
- [X] I have checked that the extension does not already exist by searching the [GitHub repository](https://github.com/tachiyomiorg/tachiyomi-extensions/) and verified it does not appear in the code base.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #16751: Paragonscans
### Source name

paragon scans

### Source link

https://paragonscans.com

### Source language

English

### Other details

English language source.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a title with source name.
- [X] I have checked that the extension does not already exist on the [website extensions list](https://tachiyomi.org/extensions/) or the app.
- [X] I have checked that the extension does not already exist by searching the [GitHub repository](https://github.com/tachiyomiorg/tachiyomi-extensions/) and verified it does not appear in the code base.
- [X] I will fill out all of the requested information in this form.

### LINKED ISSUE #17096: Mangamammy
### Source name

Manga Mammy

### Source link

https://mangamammy.ru/

### Source language

russia

### Other details

18+/nsfw= yes

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open or closed issue.
- [X] I have written a title with source name.
- [X] I have checked that the extension does not already exist on the [website extensions list](https://tachiyomi.org/extensions/) or the app.
- [X] I have checked that the extension does not already exist by searching the [GitHub repository](https://github.com/tachiyomiorg/tachiyomi-extensions/) and verified it does not appear in the code base.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by vetleledaal**
Rebasing on top of master after #17539 is merged. Feel free to suggest name(/code) changes in the meantime.

### Sites
| Name                            | URL                          | 💬 | NSFW | Note                   |
| ------------------------------- | ---------------------------- | ---- | ---- | ---------------------- |
| 1st Kiss-Manga (unoriginal)     | https://1stkiss-manga.com/   | en   | ❌   |                        |
| 1st Manhwa                      | https://1stmanhwa.com/       | en   | ✅   |                        |
| 1stKissManga.blog               | https://1stkissmanga.blog/   | en   | ✅   |                        |
| 1stKissManga.tv                 | https://1stkissmanga.tv/     | en   | ✅   |                        |
| ArcheR Scans                    | https://www.archerscans.com/ | en   | ❌   |                        |
| Asura Scans.us (unoriginal)     | https://asurascans.us/       | en   | ❌   |                        |
| Banana Manga                    | https://bananamanga.net/     | en   | ✅   |                        |
| CoffeeManga.top (unoriginal)    | https://coffeemanga.top/     | en   | ✅   |                        |
| Comic Scans                     | https://www.comicscans.org/  | en   | ❌   |                        |
| Comicz.net v2                   | https://v2.comiz.net/        | all  | ✅   | Fix #16581             |
| Elite Manga                     | https://www.elitemanga.org/  | en   | ❌   |                        |
| FactManga                       | https://factmanga.com/       | en   | ❌   |                        |
| Free Manhwa                     | https://manhwas.com/         | en   | ❌   |                        |
| Girls Love Manga!               | https://glmanga.com/         | en   | ✅   |                        |
| Grabber Zone                    | https://grabber.zone/        | all  | ❌   |                        |
| Hentai3z                        | https://hentai3z.xyz/        | en   | ✅   | Fix #14835, Fix #15788 |
| Hentai4Free                     | https://hentai4free.net/     | en   | ✅   |                        |
| Jimanga                         | https://jimanga.com/         | en   | ❌   | Fix #16224             |
| Kings-Manga                     | https://www.kings-manga.co/  | th   | ❌   |                        |
| Komik Gue                       | https://komikgue.pro/        | id   | ✅   |                        |
| Luffy Manga                     | https://luffymanga.com/      | en   | ❌   |                        |
| Manga 18h                       | https://manga18h.com/        | en   | ✅   |                        |
| Manga 18x                       | https://manga18x.net/        | en   | ✅   |                        |
| Manga Bee                       | https://mangabee.net/        | en   | ✅   |                        |
| Manga Kitsu                     | https://mangakitsu.com/      | en   | ❌   |                        |
| Manga Mammy                     | https://mangamammy.ru/       | ru   | ✅   | Fix #17096             |
| Manga Nerds                     | https://manganerds.com/      | en   | ❌   |                        |
| Manga Queen.com                 | https://mangaqueen.com/      | en   | ✅   |                        |
| Manga Queen.online (unoriginal) | https://mangaqueen.online/   | en   | ✅   |                        |
| Manga Rock.team (unoriginal)    | https://mangarock.team/      | en   | ❌   |                        |
| Manga Tx.gg (unoriginal)        | https://mangatx.gg/          | en   | ❌   |                        |
| Manga-1001.com                  | https://manga-1001.com/      | en   | ❌   |                        |
| Manga-Raw.info (unoriginal)     | https://manga-raw.info/      | en   | ✅   |                        |
| MangaClash.tv (unoriginal)      | https://mangaclash.tv/       | en   | ✅   |                        |
| MangaCrazy                      | https://mangacrazy.net/      | all  | ✅   |                        |
| MangaDino.top (unoriginal)      | https://mangadino.top/       | en   | ✅   |                        |
| MangaGo Yaoi                    | https://mangagoyaoi.com/     | en   | ✅   |                        |
| Mangakakalot.io (unoriginal)    | https://mangakakalot.io/     | en   | ✅   |                        |
| Mangakakalot.one (unoriginal)   | https://mangakakalot.one/    | en   | ✅   |                        |
| Manganelo.biz                   | https://manganelo.biz/       | en   | ✅   |                        |
| Manganelo.website (unoriginal)  | https://manganelo.website/   | en   | ✅   |                        |
| MangaOnline.team (unoriginal)   | https://mangaonline.team/    | en   | ❌   |                        |
| MangaOwl.blog (unoriginal)      | https://mangaowl.blog/       | en   | ✅   |                        |
| MangaOwl.io (unoriginal)        | https://mangaowl.io/         | en   | ✅   |                        |
| MangaOwl.one (unoriginal)       | https://mangaowl.one/        | en   | ✅   |                        |
| MangaOwl.us (unoriginal)        | https://mangaowl.us/         | en   | ✅   |                        |
| MangaPure                       | https://mangapure.net/       | en   | ✅   |                        |
| Mangareading                    | https://mangareading.org/    | en   | ✅   |                        |
| MangaRosie                      | https://mangarosie.in/       | en   | ✅   |                        |
| MangaRuby.com                   | https://mangaruby.com/       | en   | ✅   |                        |
| Mangaryu                        | https://mangaryu.com/        | en   | ✅   |                        |
| MangaTop.site                   | https://mangatop.site/       | en   | ✅   |                        |
| MangaTyrant                     | https://mangatyrant.com/     | en   | ❌   |                        |
| MangaUpdates.top (unoriginal)   | https://mangaupdates.top/    | en   | ✅   |                        |
| Manhua Zonghe                   | https://manhuazonghe.com/    | en   | ✅   |                        |
| ManhuaDex                       | https://manhuadex.com/       | en   | ❌   |                        |
| ManhuaFast.net (unoriginal)     | https://manhuafast.net/      | en   | ❌   | Fix #15277  |
| ManhuaManhwa                    | https://manhuamanhwa.com/    | en   | ✅   |                        |
| ManhuaManhwa.online             | https://manhuamanhwa.online/ | en   | ❌   |                        |
| ManhuaScan.info (unoriginal)    | https://manhuascan.info/     | en   | ✅   |                        |
| Manhwa2Read                     | https://manhwa2read.com/     | en   | ❌   |                        |
| ManhwaManhua                    | https://manhwamanhua.com/    | en   | ✅   |                        |
| ManhwaNew                       | https://manhwanew.com/       | en   | ✅   |                        |
| Mono Manga                      | https://monomanga.com/       | tr   | ❌   |                        |
| NekoPost.co (unoriginal)        | https://www.nekopost.co/     | th   | ❌   |                        |
| OnlyManhwa                      | https://onlymanhwa.org/      | en   | ✅   |                        |
| Paragon Scans                   | https://paragonscans.com/    | en   | ✅   | Fix #16751             |
| Paw Manga                       | https://pawmanga.com/        | en   | ✅   |                        |
| Pony Manga                      | https://ponymanga.com/       | en   | ✅   |                        |
| Toonizy                         | https://toonizy.com/         | en   | ✅   |                        |
| Webdex Scans                    | https://webdexscans.com/     | en   | ❌   |                        |
| Webtoon City                    | https://webtooncity.com/     | en   | ❌   |                        |
| ZinManga.top (unoriginal)       | https://zinmanga.top/        | en   | ❌   |                        |

### MadaraGenerator.kt
```kotlin
/* -- see commits -- */
```

Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [x] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [x] Have not changed source names
- [x] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio

### CODE REVIEW

**Comment by yemenihandsome** [comment]
@vetleledaal 
https://kolmanga.com

**Comment by vetleledaal** [comment]
Should be ready for review. I also sorted MadaraGenerator, so I didn't have to add the different sources manually in the right spot. This is in a separate commit so it's easy to exclude when reviewing.

**Comment by alessandrojean** [comment]
There's a few conflicts.

**Comment by vetleledaal** [comment]
Rebased and fixed conflicts. Had to fixup the code commit, as it included overrides for Grabber Zone  (oops)

File: multisrc/overrides/madara/coffeemangatop/src/CoffeeMangaTop.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    override fun chapterListParse(response: Response): List<SChapter> {
+        val document = response.asJsoup()
+        val chaptersWrapper = document.select("div[id^=manga-chapters-holder]")
+
+        var chapterElements = document.select(chapterListSelector())
+
+        if (chapterElements.isEmpty() && !chaptersWrapper.isNullOrEmpty()) {
+            val mangaId = chaptersWrapper.attr("data-id")
+            val xhrHeaders = headersBuilder()
+                .add("Referer", "$baseUrl/")
```
Comment (AwkwardPeak7) [inline multisrc/overrides/madara/coffeemangatop/src/CoffeeMangaTop.kt:40]: Madara already has the referer header added, adding here again will result in duplicate referer headers.

**Pull request merged**